src/Features/Core/Portable/SemanticSearch/Tools/ReferencingSyntaxFinder.cs

```diff
@@ -34,85 +32,44 @@ public IEnumerable<SyntaxNode> Find(ISymbol symbol)
 
     public async IAsyncEnumerable<SyntaxNode> FindAsync(ISymbol symbol)
     {
-        var channel = Channel.CreateUnbounded<ReferenceLocation>(new()
```

Comment:
Moved to helper in ProducerConsumer that already does the concurrent production/consumption work over an unbounded channel.